chainer/function_hooks/timer.py

```diff
@@ -6,56 +6,105 @@
 from chainer import function
 
 
+class Timer(object):
+
+    def __init__(self, xp):
+        self.xp = xp
+        self.running = False
+        self.reset()
+
+    def reset(self):
+        self.total_time = 0.0
+        self.last_increment = None
+        self.count = 0
+
+    def start(self):
+        if self.running:
+            return
+        if self.xp == numpy:
+            self._start = time.time()
+        else:
+            self._start = cuda.Event()
+            self._start.record()
+        self.running = True
+
+    def stop(self):
+        if not self.running:
+            return 0.0
+
+        if self.xp == numpy:
+            self._stop = time.time()
+            elapsed_time = self._stop - self._start
+        else:
+            self._stop = cuda.Event()
+            self._stop.record()
```

Comment:
I think I cannot fully understand your question. Is your question that if the kernel is launched after this record query, we cannot include this kernel to the elapse time?